src/planscape/planning/views.py

```diff
@@ -737,3 +748,45 @@ def get_treatment_goals_config_for_region(params: QueryDict):
 def treatment_goals_config(request: HttpRequest) -> HttpResponse:
     treatment_goals = get_treatment_goals_config_for_region(request.GET)
     return JsonResponse(treatment_goals, safe=False)
+
+
+#### SHARED LINK Handlers ####
+def get_shared_link(request: HttpRequest, link_code: str) -> HttpResponse:
+    try:
+        link_obj = SharedLink.objects.get(link_code=link_code)
+    except SharedLink.DoesNotExist:
+        # Handle the case where the object doesn't exist
+        raise Http404("This link does not exist")
+
+    return JsonResponse(link_obj.view_state, safe=False)
+
+
+def create_shared_link(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+
+        # read the JSON config that stores the frontend state
+        body = json.loads(request.body)
+        view_state = body.get("view_state")
```

Comment:
I'm open to not using an attribute called "view_state", but I wanted to make it possible to include other JSON data, as well, if necessary in the future.